Specs/Scene/MaterialSpec.js

```diff
@@ -655,6 +655,206 @@ describe(
+      return pollToPromise(
+        function () {
+          renderMaterial(material, true);
+          return material._textures["image"] !== material._defaultTexture;
+        },
+        { timeout: 10000 }
+      ).then(function () {
+        renderMaterial(material, true, function (rgba) {
+          expect(rgba).toEqual([0, 255, 0, 255]);
+        });
+      });
+    });
+
+    it("handles when material image is changed from default to some image", function () {
+      var material = Material.fromType(Material.ImageType, {
+        image: Material.DefaultImageId,
+        color: Color.WHITE,
+      });
+
+      renderMaterial(material, false, function (rgba) {
+        expect(rgba).toEqual([255, 255, 255, 255]);
+      });
+
+      material.uniforms.image = "./Data/Images/Green.png";
+      return pollToPromise(
+        function () {
+          renderMaterial(material, true);
+          return material._textures["image"] !== material._defaultTexture;
+        },
+        { timeout: 10000 }
+      ).then(function () {
+        renderMaterial(material, true, function (rgba) {
+          expect(rgba).toEqual([0, 255, 0, 255]);
+        });
+      });
+    });
+
+    it("handles when material image is changed from some image to undefined", function () {
+      var material = Material.fromType(Material.ImageType, {
+        image: "./Data/Images/Green.png",
+        color: Color.WHITE,
+      });
+
+      return pollToPromise(
+        function () {
+          renderMaterial(material, true);
+          return material._textures["image"] !== material._defaultTexture;
+        },
+        { timeout: 10000 }
+      )
+        .then(function () {
+          renderMaterial(material, true, function (rgba) {
+            expect(rgba).toEqual([0, 255, 0, 255]);
+          });
+        })
+        .then(function () {
+          material.uniforms.image = undefined;
+        })
+        .then(function () {
+          renderMaterial(material, true, function (rgba) {
+            expect(rgba).toEqual([255, 255, 255, 255]);
+          });
+        });
+    });
+
+    it("handles when material image is changed from some image to default", function () {
+      var material = Material.fromType(Material.ImageType, {
+        image: "./Data/Images/Green.png",
+        color: Color.WHITE,
+      });
+
+      return pollToPromise(
+        function () {
+          renderMaterial(material, true);
+          return material._textures["image"] !== material._defaultTexture;
+        },
+        { timeout: 10000 }
+      )
+        .then(function () {
+          renderMaterial(material, true, function (rgba) {
+            expect(rgba).toEqual([0, 255, 0, 255]);
+          });
+        })
+        .then(function () {
+          material.uniforms.image = Material.DefaultImageId;
+        })
+        .then(function () {
+          renderMaterial(material, true, function (rgba) {
+            expect(rgba).toEqual([255, 255, 255, 255]);
+          });
+        });
```

Comment:
These three `.then` functions can be collapsed into a single function since there's no asynchronous code within them.

Double check this for the other added tests.

```suggestion
        .then(function () {
          renderMaterial(material, true, function (rgba) {
            expect(rgba).toEqual([0, 255, 0, 255]);
          });
          material.uniforms.image = Material.DefaultImageId;
          renderMaterial(material, true, function (rgba) {
            expect(rgba).toEqual([255, 255, 255, 255]);
          });
```